src/src/com/microsoft/aad/adal/DefaultTokenCacheStore.java

```diff
@@ -53,52 +53,57 @@
     .create();
-
     private static StorageHelper sHelper;
-
     private static Object sLock = new Object();
-    
     /**
      * @param context {@link Context}
      * @throws NoSuchAlgorithmException
      * @throws NoSuchPaddingException
      */
     public DefaultTokenCacheStore(Context context) {
+        if (context == null) {
+            throw new IllegalArgumentException("Context is null");
+        }
         mContext = context;
-        if (context != null) {
-
-            if (!StringExtensions.IsNullOrBlank(AuthenticationSettings.INSTANCE
-                    .getSharedPrefPackageName())) {
-                try {
-                    // Context is created from specified packagename in order to
-                    // use same file. Reading private data is only allowed if apps specify same
-                    // sharedUserId. Android OS will assign same UID, if they
-                    // are signed with same certificates.
-                    mContext = context.createPackageContext(
-                            AuthenticationSettings.INSTANCE.getSharedPrefPackageName(),
-                            Context.MODE_PRIVATE);
-                } catch (NameNotFoundException e) {
-                    throw new IllegalArgumentException("Package name:"
-                            + AuthenticationSettings.INSTANCE.getSharedPrefPackageName()
-                            + " is not found");
-                }
-            }
-            mPrefs = mContext.getSharedPreferences(SHARED_PREFERENCE_NAME, Activity.MODE_PRIVATE);
-            if (mPrefs == null) {
-                throw new IllegalStateException(ADALError.DEVICE_SHARED_PREF_IS_NOT_AVAILABLE.getDescription());
+        if (!StringExtensions.IsNullOrBlank(AuthenticationSettings.INSTANCE
+                .getSharedPrefPackageName())) {
+            try {
+                // Context is created from specified packagename in order to
+                // use same file. Reading private data is only allowed if apps specify same
+                // sharedUserId. Android OS will assign same UID, if they
+                // are signed with same certificates.
+                mContext = context.createPackageContext(
+                        AuthenticationSettings.INSTANCE.getSharedPrefPackageName(),
+                        Context.MODE_PRIVATE);
+            } catch (NameNotFoundException e) {
+                throw new IllegalArgumentException("Package name:"
+                        + AuthenticationSettings.INSTANCE.getSharedPrefPackageName()
+                        + " is not found");
             }
+        }
+        mPrefs = mContext.getSharedPreferences(SHARED_PREFERENCE_NAME, Activity.MODE_PRIVATE);
+        if (mPrefs == null) {
+            throw new IllegalStateException(ADALError.DEVICE_SHARED_PREF_IS_NOT_AVAILABLE.getDescription());
+        }
 
-        } else {
-            throw new IllegalArgumentException("Context is null");
+        try {
```

Comment:
because if we can't load secretkey at this point we won't be able to encrypt/decrypt anything and this cache becomes useless...